src/RwaLiquidationOracle.sol

```diff
@@ -0,0 +1,262 @@
+        ilks[ilk].doc = doc;
+        ilks[ilk].tau = tau;
+
+        if (ilks[ilk].pip == address(0)) {
+            DSValue pip = new DSValue();
+            ilks[ilk].pip = address(pip);
+            pip.poke(bytes32(val));
+        } else {
+            val = uint256(DSValue(ilks[ilk].pip).read());
+        }
+
+        emit Init(ilk, val, doc, tau);
+    }
+
+    /*//////////////////////////////////
+                 Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Performs valuation adjustment for a given ilk.
+     * @param ilk The ilk to adjust.
+     * @param val The new value.
+     */
+    function bump(bytes32 ilk, uint256 val) external auth {
+        DSValue pip = DSValue(ilks[ilk].pip);
+        require(address(pip) != address(0), "RwaOracle/unknown-ilk");
+        require(ilks[ilk].toc == 0, "RwaOracle/in-remediation");
+
+        require(val >= uint256(pip.read()), "RwaOracle/decreasing-val");
+        pip.poke(bytes32(val));
+
+        emit Bump(ilk, val);
+    }
+
+    /**
+     * @notice Enables liquidation for a given ilk.
+     * @param ilk The ilk being liquidated.
+     */
+    function tell(bytes32 ilk) external auth {
+        require(ilks[ilk].pip != address(0), "RwaOracle/unknown-ilk");
+
+        (, , , uint256 line, ) = vat.ilks(ilk);
+        require(line == 0, "RwaOracle/nonzero-line");
+
+        ilks[ilk].toc = uint48(block.timestamp);
+
+        emit Tell(ilk);
+    }
+
+    /**
+     * @notice Remediation: stops the liquidation process for a given ilk.
+     * @param ilk The ilk being remediated.
+     */
+    function cure(bytes32 ilk) external auth {
+        require(ilks[ilk].pip != address(0), "RwaOracle/unknown-ilk");
+        require(ilks[ilk].toc > 0, "RwaOracle/not-in-liquidation");
+
+        ilks[ilk].toc = 0;
+
+        emit Cure(ilk);
+    }
+
+    /**
+     * @notice Writes-off a specific urn for a given ilk.
+     * @dev It assigns the outstanding debt of the urn to the vow.
+     * @param ilk The ilk being liquidated.
+     * @param urn The urn being written-off.
+     */
+    function cull(bytes32 ilk, address urn) external auth {
+        require(ilks[ilk].pip != address(0), "RwaOracle/unknown-ilk");
+        require(block.timestamp >= DSMathCustom.add(ilks[ilk].toc, ilks[ilk].tau), "RwaOracle/early-cull");
+
+        DSValue(ilks[ilk].pip).poke(bytes32(0));
+
+        (uint256 ink, uint256 art) = vat.urns(ilk, urn);
+
+        vat.grab(ilk, urn, address(this), vow, -int256(ink), -int256(art));
+
+        emit Cull(ilk, urn);
+    }
+
+    /**
+     * @notice Allows off-chain parties to check the state of the loan.
+     * @param ilk the Ilk.
+     */
+    function good(bytes32 ilk) external view returns (bool) {
+        require(ilks[ilk].pip != address(0), "RwaOracle/unknown-ilk");
+
+        return (ilks[ilk].toc == 0 || block.timestamp < DSMathCustom.add(ilks[ilk].toc, ilks[ilk].tau));
+    }
+}
```

Comment:
I'm assuming the only changes in this file is comments?

Reply:
Comments internal organization/ordering of declarations. I kept the same test cases for the Oracle to make sure nothing was broken.

I also extracted Oracle specific test cases into their own file for better organization.